solidity/test/deployERC20.ts

```diff
@@ -1,166 +1,162 @@
+
+    const txAmounts = new Array(numTxs);
+    for (let i = 0; i < numTxs; i++) {
+        txFees[i] = 1;
+        txAmounts[i] = 1;
+        txDestinationsInt[i] = signers[i + 5];
+    }
+    const txDestinations = await getSignerAddresses(txDestinationsInt);
+    let batchNonce = 1
+    let batchTimeout = 10000
+
+    // Call method
+    // ===========
+    const methodName = ethers.utils.formatBytes32String(
+        "transactionBatch"
+    );
+    let abiEncoded = ethers.utils.defaultAbiCoder.encode(
+        [
+            "bytes32",
+            "bytes32",
+            "uint256[]",
+            "address[]",
+            "uint256[]",
+            "uint256",
+            "address",
+            "uint256"
+        ],
+        [
+            gravityId,
+            methodName,
+            txAmounts,
+            txDestinations,
+            txFees,
+            batchNonce,
+            eventArgs._tokenContract,
+            batchTimeout
+        ]
+    );
+    let digest = ethers.utils.keccak256(abiEncoded);
+    let sigs = await signHash(validators, digest);
+    let currentValsetNonce = 0;
+
+    let valset = {
+        validators: await getSignerAddresses(validators),
+        powers,
+        valsetNonce: currentValsetNonce,
+        rewardAmount: 0,
+        rewardToken: ZeroAddress
+    }
+
+    await gravity.submitBatch(
+        valset,
+
+        sigs,
+
+        txAmounts,
+        txDestinations,
+        txFees,
+        batchNonce,
+        eventArgs._tokenContract,
+        batchTimeout
+    );
 
-  // Non-duplicate & sorted validators must work
-  it("runs with non-duplicate, sorted validators", async function () {
-    await runTest({ sortValidators: true })
-  });
+    // Check that Gravity's balance is correct
+    expect((await ERC20contract.functions.balanceOf(gravity.address)).toString()).to.equal(maxUint256.sub(200).toString())
 
-  // Non-duplicate yet unsorted validators must fail
-  it("throws MalformedNewValidatorSet on non-duplicate, unsorted validators", async function () {
-    await expect(runTest({})).to.be.revertedWith(
-      "MalformedNewValidatorSet()"
-    );
-  });
+    // Check that one of the recipient's balance is correct
+    expect((await ERC20contract.functions.balanceOf(await signers[6].getAddress())).toString()).to.equal('1')
+}
 
-  // Duplicate yet unsorted validators must fail
-  it("throws MalformedNewValidatorSet on duplicate, unsorted validators", async function () {
-    await expect(runTest({ duplicateValidator: true })).to.be.revertedWith(
-      "MalformedNewValidatorSet()"
-    );
-  });
+describe("deployERC20 tests", function () {
 
-  // Duplicate validators and already sorted must fail
-  it("throws MalformedNewValidatorSet on duplicate, sorted validators", async function () {
-    await expect(runTest({ duplicateValidator: true, sortValidators: true })).to.be.revertedWith(
-      "MalformedNewValidatorSet()"
-    );
-  });
+    // Non-duplicate & sorted validators must work
```

Comment:
do we need 4 tests starting here? Don't we just test for being sorted first then test for duplications?

Reply:
The duplicate validation in the solidity contract requires the list to be sorted (costs optimization ), that's why we have 4 tests. 2 with Sorted and 2 without sorting.